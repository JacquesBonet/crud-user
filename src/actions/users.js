import USERS from "../constants/users";
import {create,read,update, del} from "./docs"


export const createUser = (user) => (create(USERS, user));
export const readUsers = () => (read(USERS));
export const updateUser = (user) => (update(USERS, user));
export const deleteUser = (user) => (del(USERS, user));
