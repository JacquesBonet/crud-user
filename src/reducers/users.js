import USERS from "../constants/users";
import crudReducer  from './docs';

const initialState = {
  isLoading: false,
  payload: [],
}


export default function users(state = initialState, action) {
  return crudReducer(state, [USERS], action);
}
