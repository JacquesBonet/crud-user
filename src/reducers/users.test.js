import users from "./users";
import {createSuccess} from "../actions/docs";
import USERS from "../constants/users";

describe('reducer', function () {
  describe('initial state', function () {
    var initialState;

    beforeEach(function () {
      initialState = users(undefined, []);
    });

    it('should have an empty users array', function () {
      expect(initialState).toEqual({"isLoading": false, "payload": []});
    });
  });

  describe('ADD_USERS_SUCCESS', function () {
    var state = {"isLoading": false, "payload": []};
    const user = {path: USERS, nom: "Bonet", prenom: "Jacques"};

    beforeEach(function () {
    });

    it('adds the user code to the array', function () {
      const action = createSuccess( USERS, user);
      state = users(state, action);
      expect(state[0]).toMatchSnapshot();;
    });
  });
});

