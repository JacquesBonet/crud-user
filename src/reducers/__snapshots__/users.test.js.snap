exports[`reducer ADD_USERS_SUCCESS adds the user code to the array 1`] = `undefined`;
